pkg/controller/networkpolicy/validate.go

```diff
@@ -36,8 +69,33 @@ var (
 	// reservedTierNames stores the set of Tier names which cannot be deleted
 	// since they are created by Antrea.
 	reservedTierNames = sets.NewString("baseline", "application", "platform", "networkops", "securityops", "emergency")
+	// vr maintains list of validators to be called during validation of
+	// Antrea-native policies and Tier resources.
+	vr = validatorRegistry{}
+	// apv is an instance of antreaPolicyValidator to validate Antrea-native
+	// policy events.
+	apv = antreaPolicyValidator{}
+	// tv is an instance of tierValidator to validate Tier resource events.
+	tv = tierValidator{}
 )
 
+// registerAntreaPolicyValidator registers an Antrea-native policy validator to
+// the resource registry.
+func registerAntreaPolicyValidator(v *validatorRegistry, a validator) {
+	v.antreaPolicyValidators = append(v.antreaPolicyValidators, a)
+}
+
+// registerTierValidator registers a Tier validator to the resource registry.
+func registerTierValidator(v *validatorRegistry, t validator) {
+	v.tierValidators = append(v.tierValidators, t)
+}
+
+// initialize the validator registry with the validators that need to be called.
+func InitializeValidators() {
+	registerAntreaPolicyValidator(&vr, &apv)
+	registerTierValidator(&vr, &tv)
```

Comment:
Any thoughts on always registering these 2 validators in the `NewNetworkPolicyValidator` "constructor` and letting the constructor take a list of additional validator objects?

Or do you think there are some cases where one may not want to use these 2 default validators?

Reply:
thanks for the suggestion.. i cannot think of a use case to turn them off .. will update shortly